realm/realm-library/src/androidTest/java/io/realm/RealmListTest.java

```diff
@@ -717,6 +717,73 @@ public void testClearNotDeleting() {
         testRealm.commitTransaction();
     }
 
+    public void testContains() {
+        Owner owner = testRealm.where(Owner.class).findFirst();
+        Dog dog = owner.getDogs().get(0);
+        assertTrue("Should contain a particular dog.", owner.getDogs().contains(dog));
+    }
+
+    /**
+     * Test to see if a particular item that does exist in the same Realm does not
+     * exist in a query that excludes said item.
+     */
+    public void testContainsSameRealmNotContained() {
+        RealmResults<Dog> dogs = testRealm.where(Dog.class)
+                .equalTo("name", "Dog 1").or().equalTo("name", "Dog 2").findAll();
+        Dog thirdDog = testRealm.where(Dog.class)
+                .equalTo("name", "Dog 3").findFirst();
+        assertFalse("Should not contain a particular dog.", dogs.contains(thirdDog));
+    }
+
+    public void testContainsNotManaged() {
+        Owner owner = testRealm.where(Owner.class).findFirst();
+        RealmList<Dog> managedDogs = owner.getDogs();
+        // Create a unmanaged RealmList
+        RealmList<Dog> unmanagedDogs
+                = new RealmList<Dog>(managedDogs.toArray(new Dog[managedDogs.size()]));
+        Dog dog = managedDogs.get(0);
+        assertTrue("Should contain a particular dog", unmanagedDogs.contains(dog));
+    }
+
+    public void testContainsNull() {
+        Owner owner = testRealm.where(Owner.class).findFirst();
+        assertFalse("Should not contain a null item.", owner.getDogs().contains(null));
+    }
+
+    /**
+     * This test requires an additional Realm to test against.
```

Comment:
This comment doesn't really say anything about what the test does. It would make more sense if it described what the test was about.